lib/pbench/test/unit/agent/task/test_copy_result_tb.py

```diff
@@ -86,6 +86,47 @@ def request_callback(request: requests.Request):
             crt.copy_result_tb("token", access)
         # If we got this far without an exception, then the test passes.
 
+    @responses.activate
+    @pytest.mark.parametrize(
+        "metadata", (["pbench.test:TEST", "pbench.sat:FOO"], (), None)
+    )
+    def test_with_metadata(self, metadata, monkeypatch, agent_logger):
+        tb_name = "test_tarball.tar.xz"
+        tb_contents = "I'm a result!"
+        access = "private"
+
+        def request_callback(request: requests.Request):
+            if metadata:
+                assert "metadata" in request.params
+                assert request.params["metadata"] == metadata
+            else:
+                assert "metadata" not in request.params
+            return HTTPStatus.OK, {}, ""
+
+        responses.add_callback(
+            responses.PUT,
+            f"{self.config.get('results', 'server_rest_url')}/upload/{tb_name}",
+            callback=request_callback,
+        )
+
+        monkeypatch.setattr(Path, "exists", lambda self: True)
+        monkeypatch.setattr(
+            Path, "open", self.get_path_open_mock(tb_name, io.StringIO(tb_contents))
+        )
+
+        crt = CopyResultTb(
+            tb_name,
+            len(tb_contents),
+            "someMD5",
+            self.config,
+            agent_logger,
+        )
+        if metadata is None:
+            crt.copy_result_tb("token")
+        else:
+            crt.copy_result_tb("token", access, metadata)
```

Comment:
Metadata and access default values are `None` so I guess we don't need an `if-else` block. We can just call `crt.copy_result_tb("token", access, metadata)`

Reply:
I could go either way on that:  while simpler code is generally nicer, having the caller provide the `None` value is slightly different from having the callee provide it by default -- it's true that, as the code is presently constructed, the result is the same, but, if the CUT were to be changed (so that `None` is still acceptable but no longer the default), it would be good if the unit test exercised both cases.  (And, in general, we shouldn't be changing the unit tests when we change the CUT.)